lib/engine/game/g_1840/game.rb

```diff
@@ -410,6 +448,91 @@ def upgrades_to?(from, to, special = false, selected_company: nil)
         def needed_exits_for_hex(hex)
           CITY_TRACK_EXITS[hex.id]
         end
+
+        def info_train_name(train)
+          names = train.names_to_prices.keys.sort
+          active_variant = (available_trains & train.variants.keys).first
+          return names.join(', ') unless active_variant
+
+          names -= [active_variant]
+          "#{active_variant}, (#{names.join(', ')})"
+        end
+
+        def info_available_train(_first_train, train)
+          !(available_trains & train.variants.keys).empty?
+        end
+
+        def info_train_price(train)
+          name_and_prices = train.names_to_prices.sort_by { |k, _v| k }.to_h
+
+          active_variant = (available_trains & train.variants.keys).first
```

Comment:
This is the third time (available_trains & train.variants.keys) is used. Should this be it's own method?